packages/legend-application-data-cube/src/stores/builder/source/LakehouseConsumerDataCubeSourceLoaderState.ts

```diff
@@ -0,0 +1,155 @@
+  V1_AdHocDeploymentDataProductOrigin,
+  V1_DataProductOriginType,
+  V1_entitlementsDataProductDetailsResponseToDataProductDetails,
+  V1_EntitlementsLakehouseEnvironmentType,
+  type PersistentDataCube,
+} from '@finos/legend-graph';
+import type { LakehouseContractServerClient } from '@finos/legend-server-lakehouse';
+import { action, makeObservable, observable } from 'mobx';
+import { LegendDataCubeSourceBuilderType } from './LegendDataCubeSourceBuilderState.js';
+import { RawLakehouseConsumerDataCubeSource } from '../../model/LakehouseConsumerDataCubeSource.js';
+
+export class LakehouseConsumerDataCubeSourceLoaderState extends LegendDataCubeSourceLoaderState {
+  adhocDPDefinition: string | undefined;
+  dataProductId: string | undefined;
+  dpLoaded: boolean;
+
+  constructor(
+    application: LegendDataCubeApplicationStore,
+    engine: LegendDataCubeDataCubeEngine,
+    alertService: DataCubeAlertService,
+    sourceData: PlainObject,
+    persistentDataCube: PersistentDataCube,
+    onSuccess: () => Promise<void>,
+    onError: (error: unknown) => Promise<void>,
+  ) {
+    super(
+      application,
+      engine,
+      alertService,
+      sourceData,
+      persistentDataCube,
+      onSuccess,
+      onError,
+    );
+    this.dpLoaded = false;
+
+    makeObservable(this, {
+      dpLoaded: observable,
+      setDpLoaded: action,
+
+      dataProductId: observable,
+      setDataProductId: action,
+    });
+  }
+
+  setDpLoaded(dpLoaded: boolean) {
+    this.dpLoaded = dpLoaded;
+  }
+
+  setDataProductId(id: string | undefined) {
+    this.dataProductId = id;
+  }
+
+  override get isValid(): boolean {
+    return this.dpLoaded;
+  }
+
+  override get label() {
+    return LegendDataCubeSourceBuilderType.LAKEHOUSE_CONSUMER;
+  }
+
+  reset() {
+    const rawSource = RawLakehouseConsumerDataCubeSource.serialization.fromJson(
+      this.sourceData,
+    );
+    if (
+      !rawSource.origin ||
+      rawSource.origin === V1_DataProductOriginType.SDLC_DEPLOYMENT
+    ) {
+      this.setDpLoaded(true);
+    } else {
+      this.setDpLoaded(false);
+    }
+    this.setDataProductId(rawSource.paths[0]);
+    this.adhocDPDefinition = undefined;
+  }
+
+  async loadAdhocDataProduct(
+    access_token: string | undefined,
+    lakehouseContractServerClient: LakehouseContractServerClient,
+  ) {
+    const dataProducts =
+      V1_entitlementsDataProductDetailsResponseToDataProductDetails(
+        await lakehouseContractServerClient.getDataProduct(
+          this.dataProductId?.split('::').pop() ?? '',
+          access_token,
+        ),
+      );
+    if (dataProducts.length > 0) {
+      const dataProduct =
+        dataProducts.length === 1
+          ? dataProducts.at(0)
+          : dataProducts
+              .filter(
+                (dp) =>
+                  Boolean(dp.lakehouseEnvironment) &&
+                  dp.lakehouseEnvironment?.type ===
+                    V1_EntitlementsLakehouseEnvironmentType.PRODUCTION,
```

Comment:
why are we selecting production here ? should we make sure the right env makes pp or producation in this case ?

Reply:
so given we are not giving an options to select production or production parallel, we pick production if either of those 2 are present, if only one env is present we pick just that one